src/lib.rs

```diff
@@ -190,57 +219,48 @@ pub fn new_wmc_params_r(weights: ocaml::List<(f64, f64)>) -> ocaml::Pointer<Rsdd
 
 // branch & bound, expected semiring items
 #[ocaml::sig]
+#[derive(ocaml::ToValue, ocaml::FromValue)]
 pub struct RsddExpectedUtility(ExpectedUtility);
 ocaml::custom!(RsddExpectedUtility);
 
 #[ocaml::sig]
 pub struct RsddWmcParamsEU(WmcParams<ExpectedUtility>);
 ocaml::custom!(RsddWmcParamsEU);
 
+
 #[ocaml::func]
-#[ocaml::sig("rsdd_bdd_ptr -> rsdd_var_label list -> int64 -> rsdd_wmc_params_e_u -> rsdd_expected_utility * rsdd_partial_model")]
-pub fn bdd_bb(
-    bdd: &'static RsddBddPtr,
-    join_vars: ocaml::List<RsddVarLabel>,
-    num_vars: u64,
-    wmc: &RsddWmcParamsEU,
-) -> (
-    ocaml::Pointer<RsddExpectedUtility>,
-    ocaml::Pointer<RsddPartialModel>,
-) {
-    let (eu, pm) = bdd.0.bb(
-        &join_vars
-            .into_linked_list()
-            .iter()
-            .map(|x| x.0)
-            .collect::<Vec<_>>(),
-        num_vars as usize,
-        &wmc.0,
-    );
-    (RsddExpectedUtility(eu).into(), RsddPartialModel(pm).into())
+#[ocaml::sig("rsdd_expected_utility -> float * float")]
+pub fn extract(
+    eu : RsddExpectedUtility
+) -> (f64, f64) {
+  let v = eu.0 ;
+  (v.0, v.1)
 }
 
+
 #[ocaml::func]
-#[ocaml::sig("rsdd_bdd_ptr -> rsdd_var_label list -> int64 -> rsdd_wmc_params_e_u -> rsdd_expected_utility * rsdd_partial_model")]
+#[ocaml::sig("rsdd_bdd_ptr -> rsdd_bdd_ptr -> rsdd_var_label list -> int64 -> rsdd_wmc_params_e_u -> rsdd_expected_utility * rsdd_partial_model")]
```

Comment:
I'm going to spin up a changelog for the package in general, but to call out - if you change the API for any existing functions, we should def mention in a PR and make this the proper semver bump!